src/Ztm.Zcoin.Rpc/ZcoinRpcClient.cs

```diff
@@ -362,70 +388,83 @@ static ushort ToNative(PropertyType type)
+            var resp = await this.client.SendCommandAsync("exodus_gettransaction", transaction.GetHash());
             var rawRefAddress = resp.Result.Value<string>("referenceaddress");
 
             return new TransactionInfomation{
                 TxId = uint256.Parse(resp.Result.Value<string>("txid")),
-                Fee = Money.Parse(resp.Result.Value<string>("fee")),
                 SendingAddress = BitcoinAddress.Create(resp.Result.Value<string>("sendingaddress"), this.client.Network),
                 ReferenceAddress = (rawRefAddress == null || rawRefAddress == string.Empty)
                     ? null : BitcoinAddress.Create(resp.Result.Value<string>("referenceaddress"), this.client.Network),
                 IsMine = resp.Result.Value<bool>("ismine"),
+                Confirmations = resp.Result.Value<int>("confirmations"),
+                Fee = Money.Parse(resp.Result.Value<string>("fee")),
+                BlockTime = resp.Result.Value<int>("blocktime"),
+                Valid = resp.Result.Value<bool>("valid"),
+                InvalidReason = resp.Result.Value<string>("invalidreason"),
                 Version = resp.Result.Value<int>("version"),
                 TypeInt = resp.Result.Value<int>("type_int"),
                 Type = resp.Result.Value<string>("type")
             };
         }
 
-        async Task<(string Payload, int PayloadSize)> GetPayloadAsync(Transaction transaction)
+        async Task<(string payload, int payloadSize)> GetPayloadAsync(Transaction transaction)
         {
             var resp = await this.client.SendCommandAsync("exodus_getpayload", new object[]{transaction.GetHash()});
 
             var payload = resp.Result.Value<string>("payload");
             var payloadSize = resp.Result.Value<int>("payloadsize");
 
-            return (payload, payloadSize);
+            return (payload: payload, payloadSize: payloadSize);
         }
 
-        async Task<Transaction> ExtractExodusTransactionAync(Transaction transaction)
+        async Task<ExodusTransaction> TryDecodeExodusTransaction(Transaction transaction)
         {
-            var infomation = await DecodeTransactionAsync(transaction);
-            var payload = await GetPayloadAsync(transaction);
+            var infomation = await GetExodusTransactionAsync(transaction);
+            if (infomation == null)
+            {
+                return null;
+            }
 
-            ExodusTransaction tx;
+            var payload = await GetPayloadAsync(transaction);
 
             try
             {
-                tx = this.encoder.Decode(infomation.SendingAddress, infomation.ReferenceAddress, Encoders.Hex.DecodeData(payload.Payload));
+                return this.encoder.Decode(infomation.SendingAddress, infomation.ReferenceAddress, Encoders.Hex.DecodeData(payload.payload));
             }
             catch (TransactionFieldException)
```

Comment:
The decode should never fail. If it fail that mean we not follow consensus. So, we should not catch this error.

Reply:
nice catch